### PR TITLE
Removed deprecated classes and element options

### LINKED ISSUE #248: Removed deprecated features that should be removed in 2.0

### PR DESCRIPTION
**Pull request by szymach**
Resolves #248, after #244, #258 and #259

### CODE REVIEW

**Comment by szymach** [comment]
@fsi-open

File: Admin/ResourceRepository/ResourceFormBuilder.php [hunk 1]
```diff
@@ -79,7 +79,7 @@ private function createPropertyPath($key)
     /**
      * @param Element $element
      * @param \FSi\Bundle\ResourceRepositoryBundle\Model\ResourceValueRepository $valueRepository
-     * @param array|\FSi\Bundle\ResourceRepositoryBundle\Repository\Resource\Type\ResourceInterface[] $resources
```
Comment (chives) [inline Admin/ResourceRepository/ResourceFormBuilder.php]: this docblock shouldn't be changed

File: Admin/ResourceRepository/ResourceFormBuilder.php [hunk 2]
```diff
@@ -100,7 +100,7 @@ private function createFormData(
     /**
      * @param Element $element
      * @param \Symfony\Component\Form\FormBuilderInterface $builder
-     * @param array|\FSi\Bundle\ResourceRepositoryBundle\Repository\Resource\Type\ResourceInterface[] $resources
+     * @param array $resources
```
Comment (chives) [inline Admin/ResourceRepository/ResourceFormBuilder.php]: this docblock shouldn't be changed

File: DependencyInjection/Configuration.php [hunk 3]
```diff
@@ -26,17 +26,6 @@ public function getConfigTreeBuilder()
         $rootNode = $treeBuilder->root('fsi_admin');
 
         $rootNode
-            ->validate()
```
Comment (chives) [inline DependencyInjection/Configuration.php]: this should stay

File: DependencyInjection/Configuration.php [hunk 4]
```diff
@@ -51,8 +40,6 @@ public function getConfigTreeBuilder()
                         ->scalarNode('index_page')->defaultValue('@FSiAdmin/Admin/index.html.twig')->end()
                         ->scalarNode('list')->defaultValue('@FSiAdmin/List/list.html.twig')->end()
                         ->scalarNode('form')->defaultValue('@FSiAdmin/Form/form.html.twig')->end()
-                        ->scalarNode('crud_list')->defaultValue('@FSiAdmin/CRUD/list.html.twig')->end()
```
Comment (chives) [inline DependencyInjection/Configuration.php]: these options should stay

File: Admin/CRUD/AbstractCRUD.php [hunk 5]
```diff
@@ -72,32 +71,12 @@ public function setDefaultOptions(OptionsResolver $resolver)
...
-            'template_crud_create' => null,
-            'template_crud_edit' => null,
-            'template_list' => function (Options $options) {
-                return $options['template_crud_list'];
-            },
-            'template_form' => function (Options $options) {
-                return $options['template_crud_edit'];
-            }
+            'template_list' => '@FSiAdmin/CRUD/list.html.twig',
+            'template_form' => '@FSiAdmin/CRUD/form.html.twig'
```
Comment (chives) [inline Admin/CRUD/AbstractCRUD.php]: these 2 options should have `null` value by default

**Comment by szymach** [comment]
@fsi-open please check without merging, I will also add here a new changelog for version 3.0.

File: spec/FSi/Bundle/AdminBundle/Finder/AdminClassFinderSpec.php [hunk 6]
```diff
@@ -13,12 +12,10 @@ function it_find_admin_classes_in_default_path()
     {
         $paths = array(__DIR__ . self::FIXTURES_BUNDLE_PATH . '/Admin');
         /* We can't just check if result is an array with following values because it might be in other order */
-        $this->findClasses($paths)->shouldHaveCount(6);
+        $this->findClasses($paths)->shouldHaveCount(4);
         $this->findClasses($paths)->shouldContain("FSi\\Bundle\\AdminBundle\\spec\\fixtures\\Admin\\CRUDElement");
         $this->findClasses($paths)->shouldContain("FSi\\Bundle\\AdminBundle\\spec\\fixtures\\Admin\\DoctrineElement");
         $this->findClasses($paths)->shouldContain("FSi\\Bundle\\AdminBundle\\spec\\fixtures\\Admin\\SimpleAdminElement");
-        $this->findClasses($paths)->shouldContain("FSi\\Bundle\\AdminBundle\\spec\\fixtures\\Admin\\DataGridAwareElement");
```
Comment (szymach) [inline spec/FSi/Bundle/AdminBundle/Finder/AdminClassFinderSpec.php]: Should I add anything instead of these two classes, like list element or something?

Comment (chives) [inline spec/FSi/Bundle/AdminBundle/Finder/AdminClassFinderSpec.php]: not necessarily

**Comment by szymach** [comment]
@fsi-open

**Comment by szymach** [comment]
@fsi-open also has fix for current master branch

File: DependencyInjection/Configuration.php [hunk 7]
```diff
@@ -52,7 +52,7 @@ public function getConfigTreeBuilder()
                         ->scalarNode('list')->defaultValue('@FSiAdmin/List/list.html.twig')->end()
                         ->scalarNode('form')->defaultValue('@FSiAdmin/Form/form.html.twig')->end()
                         ->scalarNode('crud_list')->defaultValue('@FSiAdmin/CRUD/list.html.twig')->end()
-                        ->scalarNode('crud_form')->defaultNull()->end()
+                        ->scalarNode('crud_form')->defaultValue('@FSiAdmin/CRUD/form.html.twig')->end()
```
Comment (szymach) [inline DependencyInjection/Configuration.php]: That was the value in the [configuration reference](https://github.com/fsi-open/admin-bundle/blob/master/Resources/doc/configuration.md)

Comment (chives) [inline DependencyInjection/Configuration.php]: The value you mentioned comes from default value of `templates.form` option. If you set new value for `templates.form` then `templates.crud_form` will have the same value. After your change that behavior is lost.

**Comment by szymach** [comment]
@fsi-open

**Comment by szymach** [comment]
@rn0 @JarekW

**Pull request merged**